app/controllers/api_controller/parameters.rb

```diff
@@ -41,13 +41,15 @@ def filter_param(klass)
 
       params['filter'].select(&:present?).each do |filter|
         parsed_filter = parse_filter(filter, operators)
-        parts = parsed_filter[:attr].split(".")
-        attr = parts.pop
-        unless virtual_or_physical_attribute?(target_class(klass, parts), attr)
+        *associations, attr = parsed_filter[:attr].split(".")
+        if associations.size > 1
+          raise BadRequestError, "does not support filtering of attributes more than one association away"
```

Comment:
I'd prefer "Filtering of attributes with more than one association away is not supported"